governator-core/src/main/java/com/netflix/governator/internal/PostConstructLifecycleActions.java

```diff
@@ -1,57 +1,119 @@
- * Note that this feature is implicit in LifecycleModule and therefore does not need 
- * to be added using the LifecycleFeature multibinding.
+ * Special LifecycleFeature to support @PostConstruct annotation processing.
+ * Note that this feature is implicit in LifecycleModule and therefore does not
+ * need to be added using the LifecycleFeature multibinding.
  * 
  * @author elandau
  */
-public final class PostConstructLifecycleActions extends AbstractLifecycleFeature {
-
+public final class PostConstructLifecycleActions implements LifecycleFeature {
+    private static final Logger LOG = LoggerFactory.getLogger(PostConstructLifecycleActions.class);
     public static PostConstructLifecycleActions INSTANCE = new PostConstructLifecycleActions();
-    
+
     private PostConstructLifecycleActions() {
-        
     }
-    
+
     @Override
-    protected List<LifecycleAction> getMethodActions(final Class<?> type, final Method method) {
-        if (null != method.getAnnotation(PostConstruct.class)) {
-            method.setAccessible(true);
-            return Collections.<LifecycleAction>singletonList(new LifecycleAction() {
-                @Override
-                public void call(Object obj) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
-                    method.invoke(obj);
-                }
-                
-                @Override
-                public String toString() {
-                    return new StringBuilder()
-                        .append("PostConstruct[")
-                        .append(type.getName())
-                        .append("#")
-                        .append(method.getName())
-                        .append("]")
-                        .toString();
-                }
-            });
-        }
-        return Collections.emptyList();
+    public List<LifecycleAction> getActionsForType(final Class<?> type) {
+        return TypeInspector.accept(type, new PostConstructVisitor());
     }
-    
+
     @Override
     public String toString() {
         return "PostConstruct";
     }
 
+    private static class PostConstructVisitor implements TypeVisitor, Supplier<List<LifecycleAction>> {
+        private Set<String> visitContext = new HashSet<>();
+        private LinkedList<LifecycleAction> typeActions = new LinkedList<>();
+
+        @Override
+        public boolean visit(final Class<?> clazz) {
+            return !clazz.isInterface();
+        }
+
+        @Override
+        public boolean visit(final Method method) {
+            int modifiers = method.getModifiers();
+            if (method.isAnnotationPresent(PostConstruct.class)) {
+                String methodName = method.getName();
+                if (Modifier.isStatic(modifiers)) {
+                    LOG.info("invalid static @PostConstruct method {}.{}()", new Object[] { method.getDeclaringClass().getName(), methodName });
```

Comment:
without that Object[] ,there's some slf4j compile issue that manifests only
in the CI check; complains about 'java.lang.String' not being a 'Marker'.
Search of interwebs found the new Object[]{...} solution, which seems to
fix it.  Local gradle build worked fine both before and after the fix.

On Wed, Jun 1, 2016 at 5:46 PM elandau <notifications@github.com> wrote:

> In
> governator-core/src/main/java/com/netflix/governator/internal/PostConstructLifecycleActions.java
> <https://github.com/Netflix/governator/pull/297#discussion_r65468515>:
>
> > +    private static class PostConstructVisitor implements TypeVisitor, Supplier<List<LifecycleAction>> {
> > +        private Set<String> visitContext = new HashSet<>();
> > +        private LinkedList<LifecycleAction> typeActions = new LinkedList<>();
> > +
> > +        @Override
> > +        public boolean visit(final Class<?> clazz) {
> > +            return !clazz.isInterface();
> > +        }
> > +
> > +        @Override
> > +        public boolean visit(final Method method) {
> > +            int modifiers = method.getModifiers();
> > +            if (method.isAnnotationPresent(PostConstruct.class)) {
> > +                String methodName = method.getName();
> > +                if (Modifier.isStatic(modifiers)) {
> > +                    LOG.info("invalid static @PostConstruct method {}.{}()", new Object[] { method.getDeclaringClass().getName(), methodName });
>
> Why the newObject[]?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Netflix/governator/pull/297/files/831626d4e6cfd260e5d9b5b8516286c977ba2e89#r65468515>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACIO-0GH5-Z4oOsaE9XJAqu_gXLQiMD1ks5qHifrgaJpZM4InDUB>
> .
>